cdap-security/src/main/java/co/cask/cdap/security/server/ExternalAuthenticationServer.java

```diff
@@ -231,11 +233,48 @@ protected void startUp() throws Exception {
   /**
    * Initializes the handlers.
    */
-  protected void initHandlers() throws Exception {
-    authenticationHandler.init();
+  private void initHandlers() throws Exception {
+    Map<String, String> handlerProps = getAuthHandlerConfigs(cConfiguration);
+    Map<String, String> secureProps = getAuthHandlerConfigs(sConfiguration);
+
+    // used by CertificateAuthenticationHandler (see CDAP-7287)
+    copyPropIfExists(secureProps, cConfiguration, "security.auth.server.ssl.truststore.path");
+    copyPropIfExists(secureProps, cConfiguration, "security.auth.server.ssl.truststore.type");
+    copyPropIfExists(secureProps, cConfiguration, "security.auth.server.ssl.truststore.password");
+    // used by AbstractAuthenticationHandler
+    copyPropIfExists(handlerProps, cConfiguration, Constants.Security.SSL_ENABLED);
+    // used by BasicAuthenticationHandler
+    copyPropIfExists(handlerProps, cConfiguration, Constants.Security.BASIC_REALM_FILE);
+    // used by BJASPIAuthenticationHandler
+    copyPropIfExists(handlerProps, cConfiguration, Constants.Security.LOGIN_MODULE_CLASS_NAME);
+
+    authenticationHandler.init(handlerProps, secureProps);
     grantAccessToken.init();
   }
 
+  private void copyPropIfExists(Map<String, String> toProps, Configuration fromConf, String property) {
+    String value = fromConf.get(property);
+    if (value != null) {
+      toProps.put(property, value);
```

Comment:
if it is defined in both places, or if SSL_ENABLED is set with the prefix and without, toProps will get silently overridden. How about we log a warning if this happens.

Also, do we really want to use the properties from cConf over those from sConf?